app/src/main/java/fr/neamar/kiss/dataprovider/SearchProvider.java

```diff
@@ -14,44 +15,53 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import fr.neamar.kiss.KissApplication;
 import fr.neamar.kiss.loader.LoadSearchPojos;
 import fr.neamar.kiss.pojo.Pojo;
 import fr.neamar.kiss.pojo.SearchPojo;
 
 public class SearchProvider extends Provider<SearchPojo> {
-    private SharedPreferences prefs;
     public static final String URL_REGEX = "^(?:[a-z]+://)?(?:[a-z0-9-]|[^\\x00-\\x7F])+(?:[.](?:[a-z0-9-]|[^\\x00-\\x7F])+)+.*$";
 
-    private static final Map<String, String> searchProviderUrls = new LinkedHashMap<>();
+    private static final Set<String> defaultSearchProviders = new HashSet<>();
     private static final Pattern p = Pattern.compile(URL_REGEX);
 
     static {
-        searchProviderUrls.put("Bing", "https://www.bing.com/search?q=");
-        searchProviderUrls.put("DuckDuckGo", "https://duckduckgo.com/?q=");
-        searchProviderUrls.put("Google", "https://encrypted.google.com/search?q=");
-        searchProviderUrls.put("Qwant", "https://www.qwant.com/?q=");
-        searchProviderUrls.put("StartPage", "https://startpage.com/do/search?language=cat=web&query=");
-        searchProviderUrls.put("Wikipedia", "https://en.wikipedia.org/wiki/");
-        searchProviderUrls.put("Yahoo", "https://search.yahoo.com/search?p=");
+        defaultSearchProviders.add("Bing|https://www.bing.com/search?q=");
+        defaultSearchProviders.add("DuckDuckGo|https://duckduckgo.com/?q=");
```

Comment:
why not define these url strings in a constants file or strings.xml though this practice was already there but we can improve from that